### PR TITLE
[MM-33662] - Mock out CWS in enterprise code

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
Provides the option to use a mocked version of the CWS during local development.

#### Ticket Link
[MM-33662](https://mattermost.atlassian.net/browse/MM-33662?atlOrigin=eyJpIjoiY2FjMzIxYzJhYWI2NDQwNDhiMzUwNTg3ZDdmZDQ1MGIiLCJwIjoiaiJ9)

#### Related PRs
[Enterprise](https://github.com/mattermost/enterprise/pull/951)

#### Release Note

```release-note
Added a way to turn on use of mocked CWS through ldflags and config.mk file
```

### CODE REVIEW

File: config.mk [hunk 1]
```diff
@@ -22,3 +22,6 @@ RUN_SERVER_IN_BACKGROUND ?= true
 #
 # Posible options: test or qa
 LDAP_DATA ?= test
+
+# Mock the CWS.
+MM_ENABLE_CWS_MOCK = false
```
Comment (ethervoid) [inline config.mk:27]: I'd use MM_ENABLE_CWS_MOCK ?= false so by default is false but could be overridden by passing the env. variable

**Review by ethervoid — CHANGES_REQUESTED** [review]

File: model/cloud.go [hunk 2]
```diff
@@ -13,6 +13,8 @@ const (
 	InviteLimitation               = "invite"
 )
 
+var MockCWS string
```
Comment (ethervoid) [inline model/cloud.go:16]: Did you check using boolean instead of string?

Comment (AGMETEOR) [reply model/cloud.go:16]: Hello @ethervoid , ```-X``` works only for strings

Comment (ethervoid) [reply model/cloud.go:16]: Right!

**Review by ethervoid — APPROVED** [review]
LGTM! Great addition Allan!

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Pull request merged**